aiidalab_widgets_base/computational_resources.py

```diff
@@ -197,29 +197,22 @@ def refresh(self, _=None):
 
     @traitlets.validate("value")
     def _validate_value(self, change):
-        """If code is provided, set it as it is. If code's label is provided,
-        select the code and set it."""
-        code = change["value"]
+        """If code is provided, set it as it is. If code's UUID is provided,
+        check it is valid in DB"""
+        code_uuid = change["value"]
         self.output.value = ""
 
         # If code None, set value to None.
-        if code is None:
+        if code_uuid is None:
             return None
 
-        if isinstance(code, str):  # Check code by label.
-            if code in self.codes:
-                return self.codes[code]
-            self.output.value = f"""No code named '<span style="color:red">{code}</span>'
-            found in the AiiDA database."""
-        elif isinstance(code, orm.Code):  # Check code by value.
-            label = self._full_code_label(code)
-            if label in self.codes:
-                return code
-            self.output.value = f"""The code instance '<span style="color:red">{code}</span>'
-            supplied was not found in the AiiDA database."""
-
-        # This place will never be reached, because the trait's type is checked.
-        return None
+        try:
+            _ = orm.load_code(code_uuid)
```

Comment:
To my knowledge, the widget will be blocked while the validation is running. Therefore, I don't think we should run code as part of the validation that might require a significant amount of time (like a database query). I think the validation should not extend beyond very basic checks in the sense of whether the widget will enter a valid state with the provided value.

Anything beyond that should be handled as part of the business logic of the application. That means, we should inform consumers that the selected value does not correspond to a valid code and then allow those consumers to react to that in an appropriate manner.

Reply:
I see, thanks! The UUID is got from an instance queried from the database, to be honest, I see less point to check its existence. 
Changed and please check the last commit.